flixel/addons/display/FlxClothSprite.hx

```diff
@@ -0,0 +1,518 @@
+	 */
+	public var heightInTiles(default, null):Float;
+	/**
+	 * An array containing all vertices of the mesh, they are indexed row by row.
+	 */
+	public var points(default, null):Array<ClothPoint> = [];
+	/**
+	 * An array containing all vertices conections.
+	 */
+	public var constraints(default, null):Array<ClothConstraint> = [];
+	
+	/**
+	 * Mesh arrays. Vertices, indices, uvtData and colors to drawTriangles().
+	 */
+	private var _vertices:DrawData<Float>;
+	private var _indices:DrawData<Int>;
+	private var _uvtData:DrawData<Float>;
+	public var colors:DrawData<Int>;
+	
+	/**
+	 * Use to offset the drawing position of the mesh.
+	 */
+	private var _drawOffset:Point;
+	/**
+	 * The actual Flash BitmapData object representing the current display state of the modified framePixels.
+	 */
+	public var meshPixels(default, null):BitmapData;
+	
+	/**
+	 * Creates a FlxClothSprite at a specified position with a specified one-frame graphic. 
+	 * If none is provided, a 16x16 image of the HaxeFlixel logo is used.
+	 * 
+	 * @param	X				The initial X position of the sprite.
+	 * @param	Y				The initial Y position of the sprite.
+	 * @param	SimpleGraphic	The graphic you want to display (OPTIONAL - for simple stuff only, do NOT use for animated images!).
+	 * @param	Columns			Number of columns of the created mesh.
+	 * @param	Rows			Number of rows of the created mesh.
+	 * @param	PinnedSide		The pinned side that points are not affected by wind or velocity. Use FlxObject.UP, DOWN, LEFT, RIGHT, NONE, ANY, etc.
+	 */
+	public function new(?X:Float = 0, ?Y:Float = 0, ?SimpleGraphic:FlxGraphicAsset, Columns:Int = 0, Rows:Int = 0, PinnedSide:Int = FlxObject.UP, CrossingConstraints:Bool = false) 
+	{
+		super(X, Y, SimpleGraphic);
+		
+		this.pinnedSide = PinnedSide;
+		this.rows = Std.int(Math.max(2, Rows));
+		this.columns = Std.int(Math.max(2, Columns));
+		this.crossingConstraints = CrossingConstraints;
+		
+		_drawOffset = new Point();
+		setMesh(columns, rows);
+	}
+	
+	/**
+	 * WARNING: This will remove this sprite entirely. Use kill() if you want to disable it temporarily only and reset() it later to revive it.
+	 * Used to clean up memory.
+	 */
+	override public function destroy():Void
+	{
+		points = null;
+		constraints = null;
+		_vertices = null;
+		_indices = null;
+		_uvtData = null;
+		colors = null;
+		_drawOffset = null;
+		
+		meshVelocity = FlxDestroyUtil.put(meshVelocity);
+		meshFriction = FlxDestroyUtil.put(meshFriction);
+		meshPixels = FlxDestroyUtil.dispose(meshPixels);
+		
+		super.destroy();
+	}
+	
+	/**
+	 * Core update loop
+	 */
+	override public function update(elapsed:Float):Void
+	{
+		updatePoints(elapsed);
+		for (i in 0...iterations) 
+		{
+			updateConstraints(elapsed);
+		}
+		
+		super.update(elapsed);
+	}
+	
+	/**
+	 * Called by game loop, updates then blits or renders current frame of animation to the screen.
+	 */
+	override public function draw():Void 
```

Comment:
Please check if it's sufficient to override `drawComplex()`. This creates a lot of redundant code.